### ISSUE TITLE
Roadmap v5

### ISSUE THREAD

**Issue by ThomasBleijendaal** [ThomasBleijendaal/RapidCMS#182]
- [ ] Replace all interactions and dispatchers with mediator handlers
- [ ] Replacing own mediator with MediatR
- [ ] Simplify EditContext

(more)

**Comment by ljfio** [comment]
Are you looking to remove the `ModelMaker` from the solution in v5?

Also I would be willing to contribute moving to the source generated [Mediator](/martinothamar/Mediator) package?

**Comment by ThomasBleijendaal** [comment]
@ljfio Yeah, I think I am removing the ModelMaker stuff. On one hand it would have been nice to have something like that, but implementing and using it is a bit hard, and writing a DbContext from scratch is not that hard. I think I can better improve the convention-based collection logic, or make it that easily extendible. 

The source generated mediator package looks interesting, do you think it would fit within RapidCMS? I use my own mediator as a way to handle events from the UI to other parts of the UI, and use interaction and dispatcher stuff like sort of request handlers. It would be nice if everything can be replace by one mediator setup, so its more simple, and extensible by adding / replacing handlers.

**Comment by TheColonel2688** [comment]
Were these ever completed?

**Comment by ThomasBleijendaal** [comment]
I've somewhat of a made a start ([see this diff for the commits for new editors if you're interested in those](https://github.com/ThomasBleijendaal/RapidCMS/compare/master...package-short)), but focused on other projects first.
